### PR TITLE
Housecleaning: move to dune and get rid of compilation warnings

### PR DESCRIPTION
**Pull request by anmonteiro**

### CODE REVIEW

File: graphql/src/dune [hunk 1]
```diff
@@ -0,0 +1,6 @@
+(library
+ (name graphql)
+ (public_name graphql)
+ (wrapped false)
+ (libraries graphql_parser yojson rresult seq)
+ (flags (:standard -w -30-9-32)))
```
Comment (rrosenzweig) [inline graphql/src/dune]: Can `-9` and `-32` be omitted?

File: graphql/src/graphql_schema.ml [hunk 2]
```diff
@@ -72,12 +72,11 @@ module Make (Io : IO) (Stream: Stream with type 'a io = 'a Io.t) = struct
...
           bind (f x) (fun x' -> map_s ~memo:(x'::memo) f xs)
 
-    let rec map_p f xs =
-      List.map f xs |> all
+    let map_p f xs = List.map f xs |> all
 
     module Infix = struct
       let (>>=) = bind
-      let (>>|) = map
+      let (>>|) x f = map x ~f
```
Comment (rrosenzweig) [inline graphql/src/graphql_schema.ml]: This doesn't appear to be necessary?

File: graphql_parser/src/dune [hunk 3]
```diff
@@ -0,0 +1,7 @@
+(library
+ (name graphql_parser)
+ (public_name graphql_parser)
+ (wrapped false)
+ (preprocess (pps ppx_sexp_conv))
+ (libraries angstrom sexplib)
+ (flags (:standard -w -30-32)))
```
Comment (rrosenzweig) [inline graphql_parser/src/dune]: Can `-32` be omitted?

File: graphql/src/graphql_schema.ml (hunk 2 above)
Comment (anmonteiro) [reply graphql/src/graphql_schema.ml]: If I don't do this, the compiler emits:

> Error (warning 6): label f was omitted in the application of this function.

**Comment by anmonteiro** [comment]
Pushed a commit addressing the code review items, but note that eliminating warning 32 means I deleted some unused declarations.

**Pull request merged**